modules/SofaExporter/CMakeLists.txt

```diff
@@ -44,10 +44,23 @@ list(APPEND SOURCE_FILES
 
 add_library(${PROJECT_NAME} SHARED ${HEADER_FILES} ${SOURCE_FILES} ${EXTRA_FILES})
 target_link_libraries(${PROJECT_NAME} PUBLIC SofaSimulationTree)
-target_compile_definitions(${PROJECT_NAME} PRIVATE COMPILE_FLAGS "-DSOFA_BUILD_EXPORTER")
-target_compile_definitions(${PROJECT_NAME} PUBLIC COMPILE_FLAGS "-DSOFA_HAVE_SOFAEXPORTER")
-
+target_compile_definitions(${PROJECT_NAME} PRIVATE "-DSOFA_BUILD_EXPORTER")
+target_compile_definitions(${PROJECT_NAME} PUBLIC "-DSOFA_HAVE_SOFAEXPORTER")
 target_include_directories(${PROJECT_NAME} PUBLIC "$<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/src>")
 target_include_directories(${PROJECT_NAME} PUBLIC "$<INSTALL_INTERFACE:include/${PROJECT_NAME}>")
 
-sofa_generate_package(NAME SofaExporter VERSION ${PROJECT_VERSION} TARGETS ${PROJECT_NAME} INCLUDE_INSTALL_DIR ${PROJECT_NAME} INCLUDE_SOURCE_DIR "src/SofaExporter")
+sofa_generate_package(
+    NAME SofaExporter
+    VERSION ${PROJECT_VERSION}
+    TARGETS ${PROJECT_NAME}
+    INCLUDE_SOURCE_DIR "src/SofaExporter"
+    INCLUDE_INSTALL_DIR "SofaExporter/SofaExporter"
+    )
+
+# Enable tests if SofaTest found
+# Will pass only in out-of-SOFA builds
+# See modules/tests/CMakeLists.txt if building through SOFA
+find_package(SofaTest QUIET)
+if(SofaTest_FOUND)
```

Comment:
Just a simple question..: Is that the clean way of doing now, when compiling tests in a separated module / plugin?
SOFA_BUILD_TESTS should only be used in SofaKernel/framework ?

Reply:
I thinks this is the right way as long as we have a dependency mess with SofaTest (SofaTest depends on all SOFA modules).
As my comment says, SofaTest will be found only when building out-of-SOFA. 
That's why, for "modularized" modules that are still built in-SOFA but could now also be built out-of-SOFA, we need to have their tests added both in their `CMakeLists.txt` and in `modules/tests/CMakeLists.txt`.

For plugins, we should have something like this
```
option(MYPLUGIN_BUILD_TESTS "Build tests for MyPlugin" ON)
if(MYPLUGIN_BUILD_TESTS)
    add_subdirectory(MyPlugin_test)
endif()
```
without forgetting in MyPlugin/CMakeLists.txt: `find_package(SofaTest REQUIRED)`